lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -66,6 +67,26 @@ UsdUndoRenameCommand::UsdUndoRenameCommand(const UsdSceneItem::Ptr& srcItem, con
     const UsdPrim& prim = _stage->GetPrimAtPath(_ufeSrcItem->prim().GetPath());
 
     ufe::applyCommandRestriction(prim, "rename");
+
+    // handle unique name for _newName
+    TfToken::HashSet childrenNames;
+    for (auto child : prim.GetParent().GetChildren()){
+        childrenNames.insert(child.GetName());
+    }
+    if (childrenNames.find(TfToken(_newName)) != childrenNames.end()){
+        _newName = uniqueName(childrenNames, _newName);
+    }
+
+    // names are not allowed to start to digit numbers
+    if(std::isdigit(_newName.at(0))){
+        _newName = prim.GetName();
+    }
+
+    // all special characters are replaced with `_`
+    const std::string specialChars{"~!@#$%^&*()-=+,.?`':{}|<>[]/ "};
+    std::replace_if(_newName.begin(), _newName.end(), [&](auto c){
+        return std::string::npos != specialChars.find(c);
+    }, '_');
```

Comment:
Moved all the code for making the new name unique to the constructor.  This is for sake of making the renameRedo clean and also there is no real benefit to keep calling them on every redo operation.